php-frontend/src/main/java/org/sonar/plugins/php/api/tree/expression/FunctionCallTree.java

```diff
@@ -43,13 +44,29 @@
   @Nullable
   SyntaxToken openParenthesisToken();
 
+  /**
+   * @deprecated since 3.11 . Use {@link #callArguments()} instead.
+   */
+  @Deprecated
   SeparatedList<ExpressionTree> arguments();
 
+  SeparatedList<FunctionCallArgumentTree> callArguments();
+
   /**
-   * Nullable in case of internal function call with no parenthesis
+   * Retrieves an argument based on position and name.
+   *
+   * If an argument with the given name exists, it is returned no matter the position.
+   * Else, the argument at the supplied position is returned if it exists and is not named.
+   *
+   * @since 3.11
    */
   @Nullable
+  FunctionCallArgumentTree argument(int position, String name);
```

Comment:
I would not add such a method in the public API for now:
* Its contract is not trivial
* We can probably have it as a utility method somewhere else

We can later move it to the public API if we feel it's appropriate.